### ISSUE TITLE
rewrite_tag cannot construct new tag from log message

### ISSUE THREAD

**Issue by yong-jie-gong** [fluent/fluent-bit#4917]
## Bug Report

**Describe the bug**

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
{"log":"YOUR LOG MESSAGE HERE","stream":"stdout","time":"2018-06-11T14:37:30.681701731Z"}
```
- Steps to reproduce the problem:
1) use steps in https://docs.fluentbit.io/manual/pipeline/filters/rewrite-ta
2) configuration:
  ```
[SERVICE]
Flush 1
Log_Level info



[INPUT]
NAME dummy
Dummy {"tool": "fluent", "sub": {"s1": {"s2": "bit"}}}
Tag test_tag

[FILTER]
Name rewrite_tag
Match test_tag
Rule $tool ^(fluent)$ from.$TAG.new.$tool.$sub['s1']['s2'].out false
Emitter_Name re_emitted

[OUTPUT]
Name stdout
Match from.*
```
3). fluent-bit failed to start with error message
```
[2022/02/24 06:20:16] [ info] [engine] started (pid=104)
[2022/02/24 06:20:16] [ info] [storage] version=1.1.6, initializing...
[2022/02/24 06:20:16] [ info] [storage] in-memory
[2022/02/24 06:20:16] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/02/24 06:20:16] [ info] [cmetrics] version=0.2.3
[2022/02/24 06:20:16] [error] [record accessor] syntax error, unexpected '$', expecting $end at '$tool.$sub['s1']['s2'].out'
[2022/02/24 06:20:16] [error] [filter:rewrite_tag:rewrite_tag.0] could not compose tag: from.$TAG.new.$tool.$sub['s1']['s2'].out
[2022/02/24 06:20:16] [error] Failed initialize filter rewrite_tag.0
[2022/02/24 06:20:16] [error] [lib] backend failed
```
**Expected behavior**
As described in link https://docs.fluentbit.io/manual/pipeline/filters/rewrite-tag, rewrite_tag should be able to change the tag from log message, it works in 1.8.12, but broken from 1.9.0* and master

**Screenshots**

**Your Environment**

* Version used: v1.9.0-rc2
* Configuration: 
* Environment name and version (e.g. Kubernetes? What version?): 
* Server type and version: 
* Operating System and version: Redhat8
* Filters and plugins:

**Additional context**

**Comment by nokute78** [comment]
The commit https://github.com/fluent/fluent-bit/commit/9dd1cbc6527b0f5cd162c0885d36e3dbfd44de76 causes breaking change.
It is to fix #4370 and I think record accessor needs to be refactored to fix both.

**Comment by gpetrovgeorgi** [comment]
Hello I still get the same error, is this final fixed or ?
